CRM/Contribute/Form/ContributionBase.php

```diff
@@ -574,9 +574,16 @@ public function assignToTemplate() {
     }
     $this->assignPaymentFields();
 
-    $this->assign('email',
-      $this->controller->exportValue('Main', "email-{$this->_bltID}")
-    );
+    if ($this->_emailExists && empty($this->_params["email-{$this->_bltID}"])) {
+      foreach ($this->_params as $key => $val) {
+        if (substr($key, 0, 6) == 'email-') {
+          $this->assign('email', $this->_params[$key]);
+        }
+      }
+    }
+    else {
+      $this->assign('email', CRM_Utils_Array::value("email-{$this->_bltID}", $this->_params));
+    }
```

Comment:
@jitendrapurohit Can you extract this bit of code into it's own function and provide a comment explaining why we have to do it this way?

Reply:
Done.